src/components/App/App.jsx

```diff
@@ -3,22 +3,66 @@ import Backdrop from '../Backdrop/Backdrop';
 import ModalSS from '../ModalSS/ModalSS';
 import ModalGal from '../ModalGal/ModalGal';
 import Hero from '../Hero/Hero';
+import dummyData from '../../../dummyData';
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      testImgSrc: 'https://s3-us-west-1.amazonaws.com/airbnbusy/photos/p00020.jpg',
+      showModalGallery: false,
+      showModalShareSave: false,
     };
+    this.modalGalleryHandler = this.modalGalleryHandler.bind(this);
+    this.modalShareSaveHandler = this.modalShareSaveHandler.bind(this);
+    this.keyPressHandler = this.keyPressHandler.bind(this);
+  }
+
+  modalGalleryHandler() {
+    this.setState(prevState => ({ showModalGallery: !prevState.showModalGallery }));
+  }
+
+  modalShareSaveHandler(e) {
+    e.stopPropagation();
+    this.setState(prevState => ({ showModalShareSave: !prevState.showModalShareSave }));
```

Comment:
Didn't know setState can take a function as it's first argument - good to know!

Reply:
Sure can.  This pic explains it better than I would:

<img width="1179" alt="screen shot 2018-06-12 at 1 37 41 pm" src="https://user-images.githubusercontent.com/17700507/41315969-ea2ba092-6e45-11e8-86a8-9c5b5721bc31.png">